src/sentry/api/endpoints/project_dynamic_sampling.py

```diff
@@ -281,37 +341,108 @@ def get(self, request: Request, project) -> Response:
                     "sample_rate_distributions": None,
                     "startTimestamp": None,
                     "endTimestamp": None,
+                    "parentProjectBreakdown": [],
                 }
             )
-        sample_size = len(root_transactions)
+        sample_size = len(transactions)
+        # TODO: (andrii) recheck it later if we want to filter by is_root_transaction == 1
         sample_rates = sorted(
-            transaction.get("trace.client_sample_rate") for transaction in root_transactions
+            transaction.get("trace.client_sample_rate") for transaction in transactions
         )
         non_null_sample_rates = sorted(
             float(sample_rate) for sample_rate in sample_rates if sample_rate not in {"", None}
         )
 
         project_breakdown = None
+        parent_trace_ids = None
+        parent_project_breakdown = []
         if distributed_trace:
             # If the distributedTrace flag was enabled, then we are also interested in fetching
             # the project breakdown of the projects in the trace of the root transaction
-            trace_id_list = [transaction.get("trace") for transaction in root_transactions]
+
+            trace_ids = [transaction.get("trace") for transaction in transactions]
+
             projects_in_org = Project.objects.filter(organization=project.organization).values_list(
                 "id", flat=True
             )
 
-            project_breakdown = self.__run_discover_query(
-                columns=["project_id", "project", "count()"],
-                query=f"event.type:transaction trace:[{','.join(trace_id_list)}]",
-                params={
-                    "start": query_time_range.start_time,
-                    "end": query_time_range.end_time,
-                    "project_id": list(projects_in_org),
-                    "organization_id": project.organization.id,
-                },
-                limit=20,
-                referrer="dynamic-sampling.distribution.fetch-project-breakdown",
+            # Discover query to fetch parent projects (smart query)
+            projects_counts = self.__projects_counts_query(project, query_time_range, trace_ids)
+
+            # Requesting for a project that is root for some transactions,
+            # but not root for others and part of distributed trace like
+            root_projects_count = len(
+                [project for project in projects_counts if project["root"] > 0]
             )
+            # get requested project id in projects_count dataset
+            try:
+                requested_project = next(
+                    item for item in projects_counts if item["project_id"] == project.id
+                )
+            except StopIteration:
```

Comment:
This is probably redundant because we are filtering by project_id to fetch random transactions